app/models/machine.rb

```diff
@@ -164,6 +166,33 @@ def update_details_by_api(params, machine_details)
     machine_details.update(params)
   end
 
+  # try to detect minimal changes triggered by two adapters delivering
+  # the same information with slight changes (amount of RAM < 5%)
+  # or a VM detected on two different hosts (DRBD or one shut down)
+  def flapping_detection
+    begin
+      v = versions.last
+      if (v.changeset.keys.size == 1)
+        # if only one attribute has changed
+        lastkey = v.changeset.keys.last
+        if (lastkey == "ram")
+          ram1 = v.changeset.values.first.first
+          ram2 = v.changeset.values.first.last
+          if (!ram1.blank? && !ram2.blank?)
+            percentage = ((ram1.to_f - ram2.to_f)/ram1*100).abs
+            # difference must be less than 5%
+            v.destroy if (percentage < 5)
+          end
+        elsif (lastkey == "vmhost")
+          # vmhost change is the same as two versions earlier -> flapping
+          v.destroy if (v.changeset == m.versions[-3].changeset)
```

Comment:
maybe this should be `if m.versions.size > 2 and (v.changeset = m.versions[-3].changeset)` or something similar to guard for nil errors.